File: benchmarks/stable_benchmark.py
#!/usr/bin/env python3
# Stable benchmark for Python - equivalent to Sentra stable benchmark

import time

print("=== PYTHON STABLE BENCHMARK ===")
print("")

# Test 1: Arithmetic Operations (10k iterations)
print("Test 1: Arithmetic operations...")
start_time = time.time() * 1000

sum_val = 0
for i in range(10000):
    sum_val = sum_val + i * 2 - 1
    sum_val = sum_val / 1.1
    sum_val = sum_val % 10000

arithmetic_time = time.time() * 1000 - start_time
print(f"Arithmetic (10k ops): {arithmetic_time:.0f}ms")

# Test 2: Array Operations
print("")
print("Test 2: Array operations...")
start_time = time.time() * 1000

arr = []
for i in range(1000):
    arr.append(i)
    arr.append(i * 2)

total = 0
for val in arr:
    total = total + val

array_time = time.time() * 1000 - start_time
print(f"Array (2k elements): {array_time:.0f}ms")

# Test 3: Map Operations
print("")
print("Test 3: Map operations...")
start_time = time.time() * 1000

map_data = {}
for i in range(500):
    map_data[f"key{i}"] = i * 10
    map_data[f"data{i}"] = i * 20

map_count = 0
for key in map_data:
    map_count = map_count + 1

map_time = time.time() * 1000 - start_time
print(f"Map (1k entries): {map_time:.0f}ms")

# Test 4: String Operations
print("")
print("Test 4: String operations...")
start_time = time.time() * 1000

text = "benchmark"
for i in range(200):
    text = text + f" test {i}"
    if len(text) > 10000:
        text = "reset"

string_time = time.time() * 1000 - start_time
print(f"String (200 concatenations): {string_time:.0f}ms")

# Test 5: Function Calls
print("")
print("Test 5: Function calls...")

def calculate(a, b):
    return a * b + (a - b)

start_time = time.time() * 1000

func_result = 0
for i in range(2000):
    func_result = func_result + calculate(i, i + 1)

function_time = time.time() * 1000 - start_time
print(f"Function calls (2k): {function_time:.0f}ms")

# Test 6: Nested Loops
print("")
print("Test 6: Nested loops...")
start_time = time.time() * 1000

nested_sum = 0
for i in range(50):
    for j in range(50):
        nested_sum = nested_sum + i * j

nested_time = time.time() * 1000 - start_time
print(f"Nested loops (2.5k iterations): {nested_time:.0f}ms")

# Total time
total_time = arithmetic_time + array_time + map_time + string_time + function_time + nested_time
print("")
print("=== BENCHMARK COMPLETE ===")
print(f"Total execution time: {total_time:.0f}ms")
print("Platform: Python (CPython)")

File: benchmarks/stable_benchmark.js
// Stable benchmark for Node.js - equivalent to Sentra stable benchmark

console.log("=== NODE.JS STABLE BENCHMARK ===");
console.log("");

// Test 1: Arithmetic Operations (10k iterations)
console.log("Test 1: Arithmetic operations...");
let startTime = Date.now();

let sum = 0;
for (let i = 0; i < 10000; i++) {
    sum = sum + i * 2 - 1;
    sum = sum / 1.1;
    sum = sum % 10000;
}

let arithmeticTime = Date.now() - startTime;
console.log(`Arithmetic (10k ops): ${arithmeticTime}ms`);

// Test 2: Array Operations
console.log("");
console.log("Test 2: Array operations...");
startTime = Date.now();

let arr = [];
for (let i = 0; i < 1000; i++) {
    arr.push(i);
    arr.push(i * 2);
}

let total = 0;
for (let val of arr) {
    total = total + val;
}

let arrayTime = Date.now() - startTime;
console.log(`Array (2k elements): ${arrayTime}ms`);

// Test 3: Map Operations
console.log("");
console.log("Test 3: Map operations...");
startTime = Date.now();

let mapData = {};
for (let i = 0; i < 500; i++) {
    mapData[`key${i}`] = i * 10;
    mapData[`data${i}`] = i * 20;
}

let mapCount = 0;
for (let key in mapData) {
    mapCount = mapCount + 1;
}

let mapTime = Date.now() - startTime;
console.log(`Map (1k entries): ${mapTime}ms`);

// Test 4: String Operations
console.log("");
console.log("Test 4: String operations...");
startTime = Date.now();

let text = "benchmark";
for (let i = 0; i < 200; i++) {
    text = text + ` test ${i}`;
    if (text.length > 10000) {
        text = "reset";
    }
}

let stringTime = Date.now() - startTime;
console.log(`String (200 concatenations): ${stringTime}ms`);

// Test 5: Function Calls
console.log("");
console.log("Test 5: Function calls...");

function calculate(a, b) {
    return a * b + (a - b);
}

startTime = Date.now();

let funcResult = 0;
for (let i = 0; i < 2000; i++) {
    funcResult = funcResult + calculate(i, i + 1);
}

let functionTime = Date.now() - startTime;
console.log(`Function calls (2k): ${functionTime}ms`);

// Test 6: Nested Loops
console.log("");
console.log("Test 6: Nested loops...");
startTime = Date.now();

let nestedSum = 0;
for (let i = 0; i < 50; i++) {
    for (let j = 0; j < 50; j++) {
        nestedSum = nestedSum + i * j;
    }
}

let nestedTime = Date.now() - startTime;
console.log(`Nested loops (2.5k iterations): ${nestedTime}ms`);

// Total time
let totalTime = arithmeticTime + arrayTime + mapTime + stringTime + functionTime + nestedTime;
console.log("");
console.log("=== BENCHMARK COMPLETE ===");
console.log(`Total execution time: ${totalTime}ms`);
console.log("Platform: Node.js (V8)");